Always_return_checker.py

```diff
@@ -0,0 +1,80 @@
+"""checker for always returning in a loop.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class AlwaysReturnChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'always_return'
+    msgs = {'E9996': ('Always returning an object or none in the loop on line %s',
+                      'always_returning_in_a_loop',
+                      'Used when you always return an object or none in a loop, '
+                      'this may cause the loop only runs once.'),
+           }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("always_returning_in_a_loop")
+    def helper_fuc(self, node):
+        # Check if the node is a return statement
+        if isinstance(node, astroid.Return):
+            args = "{}".format(node.lineno)
+            self.add_message("always_returning_in_a_loop", node=node, args=args)
+        # check if the current branch is astroid.If
+        if isinstance(node, astroid.If):
+            flag = True
+            if not any([isinstance(child, astroid.Return) for child in (
```

Comment:
Rather than checking if the child is an `astroid.Return`, you should make *additional* recursive calls on the children. This will allow you to handle, for example, a nested `if`.